synthesizer/src/vm/verify.rs

```diff
@@ -229,11 +242,11 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
     /// Note: This is an internal check only. To ensure all components of the execution are checked,
     /// use `VM::check_transaction` instead.
     #[inline]
-    fn check_execution_internal(&self, execution: &Execution<N>) -> Result<()> {
+    fn check_execution_internal(&self, execution: &Execution<N>, already_verified: bool) -> Result<()> {
         let timer = timer!("VM::check_execution");
 
-        // Verify the execution.
-        let verification = self.process.read().verify_execution(execution);
+        // Verify the execution if it hasn't been done yet.
+        let verification = if already_verified { Ok(()) } else { self.process.read().verify_execution(execution) };
```

Comment:
This means we will skip the state root check if `already_verified`, please check all cases in this comment again: https://github.com/AleoHQ/snarkVM/pull/2270#discussion_r1440240416

Reply:
Nevermind I see we don't return